### PR TITLE
Allow lowercase `type` on config entries

### PR DESCRIPTION
**Pull request by aq17**
Addresses #201

### CODE REVIEW

File: CHANGELOG_PENDING.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 ### Improvements
 
+- Allow lowercase `type` on config entries.
+  [#201](https://github.com/pulumi/pulumi-yaml/pull/201)
```
Comment (dixler) [inline CHANGELOG_PENDING.md:4]: we use the pull request number rather than the issue number. I'm not sure why we do, but we do.

**Review by dixler — CHANGES_REQUESTED** [review]
How are your changes already in the `main` branch?

**Comment by aq17** [comment]
> How are your changes already in the `main` branch?

I have no idea 😟

**Comment by AaronFriel** [comment]
🤞

File: pkg/pulumiyaml/config/config.go [hunk 2]
```diff
@@ -80,8 +80,8 @@ func (c Types) String() string {
 }
 
 func Parse(s string) (Type, bool) {
-	if strings.HasPrefix(s, "List<") && strings.HasSuffix(s, ">") {
-		innerString := strings.TrimSuffix(strings.TrimPrefix(s, "List<"), ">")
+	if strings.HasPrefix(s, strings.ToLower("List<")) && strings.HasSuffix(s, ">") {
+		innerString := strings.TrimSuffix(strings.TrimPrefix(strings.TrimPrefix(s, "List<"), "list<"), ">")
```
[the remark this answers is not in the document]
Comment (aq17) [reply pkg/pulumiyaml/config/config.go]: Just realized that 😅

[the remark this answers is not in the document]
Comment (aq17) [reply pkg/pulumiyaml/config/config.go]: We lowercase s, and then only check/ trim `list<`

**Review by AaronFriel — APPROVED** [review]
LGTM, one nit: I'd change lines 94, 96, 98 to just have the lowercase versions as `s` will be lowercase.

**Pull request merged**